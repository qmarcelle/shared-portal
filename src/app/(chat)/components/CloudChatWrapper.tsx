'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
