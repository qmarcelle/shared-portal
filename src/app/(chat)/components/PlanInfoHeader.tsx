'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
