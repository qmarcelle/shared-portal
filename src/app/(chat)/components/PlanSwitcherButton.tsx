'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
