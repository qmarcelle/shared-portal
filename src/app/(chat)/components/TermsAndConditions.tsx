'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
