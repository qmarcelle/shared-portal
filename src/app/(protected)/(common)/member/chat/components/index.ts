/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
