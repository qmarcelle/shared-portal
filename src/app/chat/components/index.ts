// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
