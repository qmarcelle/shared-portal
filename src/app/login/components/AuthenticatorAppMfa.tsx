import { AppLink } from '@/components/foundation/AppLink';
import { Button } from '@/components/foundation/Button';
import { Spacer } from '@/components/foundation/Spacer';
import { TextField } from '@/components/foundation/TextField';
import { AppProg } from '../models/app/app_prog';
import { MfaModeState } from '../models/app/mfa_mode_state';
import { useLoginStore } from '../stores/loginStore';
import { useMfaStore } from '../stores/mfaStore';

export const AuthenticatorAppMfa = () => {
  const code = useMfaStore((state) => state.code);
  const completeMfaProg = useMfaStore((state) => state.completeMfaProg);
  const actions = useMfaStore((state) => ({
    submitMfa: state.submitMfaAuth,
    updateCode: state.updateCode,
    resendMfa: state.resendMfa,
    updateMfaStage: state.updateMfaStage,
  }));
  const apiErrors = useLoginStore((state) => state.apiErrors);

  function validateSecurityCode() {
    if (code.length > 0) {
      return () => actions.submitMfa();
    } else {
      return undefined;
    }
  }
  return (
    <div id="mainSection">
      <h1>Let&apos;s Confirm Your Identity</h1>
      <p className="mb-4">
        Enter the security code from your authenticator app.
      </p>
      <TextField
        label="Enter Security Code"
        valueCallback={(val) => actions.updateCode(val)}
        errors={apiErrors}
      />
      <Spacer size={24} />
      <Spacer size={32} />
      <Button
        callback={validateSecurityCode()}
        label={completeMfaProg == AppProg.loading ? 'Confirming' : 'Confirm'}
      />
      <Spacer size={16} />
      <AppLink
        label="Choose a Different Method"
        callback={() => actions.updateMfaStage(MfaModeState.selection)}
        className="m-auto"
      />
      <Spacer size={65} />
      <h3>Need Help?</h3>
      <p>
        Give us a call using the number listed on the back of your Member ID
        card or{' '}
        <AppLink url="https://www.bcbst.com/contact-us" label="contact us" />
      </p>
    </div>
  );
};
