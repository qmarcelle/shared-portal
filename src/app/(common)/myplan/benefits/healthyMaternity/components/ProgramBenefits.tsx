/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
