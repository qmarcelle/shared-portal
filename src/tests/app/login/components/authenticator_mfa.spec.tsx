import { AuthenticatorAppMfa } from '@/app/login/components/AuthenticatorAppMfa';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

jest.setTimeout(30000);

const setupUI = () => {
  const component = render(<AuthenticatorAppMfa />);

  const inputSecurityCode = screen.getByRole('textbox', {
    name: /Enter Security Code/i,
  });
  const resendCode = screen.getByRole('button', { name: /Resend Code/i });
  const confirmButton = screen.getByRole('button', { name: /Confirm/i });
  const chooseDiffrentMethod = screen.getByRole('button', {
    name: /Choose a Different Method/i,
  });
  const contactUs = screen.getByRole('button', { name: /contact us/i });
  return {
    component,
    inputSecurityCode,
    resendCode,
    confirmButton,
    chooseDiffrentMethod,
    contactUs,
  };
};

describe('Authenticator Mfa Component', () => {
  it('should render the UI correctly', () => {
    const ui = setupUI();

    expect(ui.inputSecurityCode).toBeVisible();
    expect(ui.confirmButton).toBeVisible();
    expect(ui.chooseDiffrentMethod).toBeVisible();
    expect(ui.contactUs).toBeVisible();
    expect(ui.contactUs).toBeVisible();

    expect(ui.component.container).toMatchSnapshot();
  });
});
