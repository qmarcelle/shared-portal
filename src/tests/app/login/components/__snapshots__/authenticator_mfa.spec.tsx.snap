// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticator Mfa Component should render the UI correctly 1`] = `
<div>
  <div
    id="mainSection"
  >
    <h2
      class="title-2"
    >
      Let's Confirm Your Identity
    </h2>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
    />
    <p>
      Enter the security code from your authenticator app.
    </p>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <div
      class="flex flex-col w-full text-field"
    >
      <p>
        Enter Security Code
      </p>
      <div
        class="flex flex-row items-center input   "
      >
        <input
          aria-label="Enter Security Code"
          class=""
          type="text"
          value=""
        />
        <div
          class="cursor-pointer"
        />
      </div>
      <div
        class="error-container mt-1 p-1"
      >
        <div />
      </div>
    </div>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <div
      class="flex flex-row justify-center items-center tooltip"
    >
      <span
        class="tooltiptext"
      >
        Enter a Security Code.
      </span>
      <button
        aria-label="Confirm"
        class="button-text primary text-center flex flex-row justify-center items-center min-w-fit inactive undefined"
        type="button"
      >
        Confirm
      </button>
    </div>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
    />
    <a
      style="display: block;"
      tabindex="-1"
    >
      <button
        class="flex flex-row link-container m-auto"
        style="max-width: max-content; height: auto; display: block;"
        tabindex="0"
      >
        <p
          class="link undefined"
          style="display: block;"
        >
          Choose a Different Method
        </p>
      </button>
    </a>
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <div
      class=""
      style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
    />
    <span
      style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
    />
    <h3>
      Need Help?
    </h3>
    <p>
      Give us a call using the number listed on the back of your Member ID card or
       
      <a
        href="https://www.bcbst.com/contact-us"
        style="display: inline;"
        tabindex="-1"
      >
        <button
          class="flex flex-row link-container pl-0 pt-0"
          style="max-width: max-content; height: auto; display: inline;"
          tabindex="0"
        >
          <p
            class="link undefined"
            style="display: inline;"
          >
            contact us
          </p>
        </button>
      </a>
    </p>
  </div>
</div>
`;
